### PR TITLE
Mon 4059 multiselect on servicegroup and criticities

### PR DESCRIPTION
**Pull request by adrien2612**
MON-4059

### CODE REVIEW

File: service-monitoring/src/export.php [hunk 1]
```diff
@@ -243,51 +243,48 @@
...
+        FROM services_servicegroups 
+        WHERE servicegroup_id IN ({$querySG})) 
+SQL;
+    $query = CentreonUtils::conditionBuilder($query, $servicegroupCondition);
 }
-if (isset($preferences["display_severities"])
-    && $preferences["display_severities"]
-    && isset($preferences['criticality_filter'])
-    && $preferences['criticality_filter'] != ""
+if (isset($preferences['display_severities']) &&
```
Comment (sc979) [inline service-monitoring/src/export.php]: LCA explained me that the boolean operator should be set at the beginning of the line  ;)
that's why I modified the coding style rules

File: service-monitoring/src/export.php [hunk 2]
```diff
@@ -243,51 +243,48 @@
...
+        $querySG .= ":id_" . $resultSG;
+        $mainQueryParameters[] = [
+            'parameter' => ':id_' . $resultSG,
+            'value' => (int)$resultSG,
+            'type' => PDO::PARAM_INT
+        ];
+    }
+    $servicegroupCondition = <<<SQL
+s.service_id IN (
+        SELECT DISTINCT service_id
```
Comment (sc979) [inline service-monitoring/src/export.php]: 4 or 8 spaces ? +333 ;)

File: service-monitoring/src/export.php [hunk 3]
```diff
@@ -243,51 +243,48 @@
...
+            'parameter' => ':id_' . $resultSG,
+            'value' => (int)$resultSG,
+            'type' => PDO::PARAM_INT
+        ];
+    }
+    $servicegroupCondition = <<<SQL
+s.service_id IN (
+        SELECT DISTINCT service_id 
+        FROM services_servicegroups 
+        WHERE servicegroup_id IN ({$querySG}))
```
Comment (sc979) [inline service-monitoring/src/export.php]: as the first parenthesis stayed on line 260, the last one should go to next line

File: service-monitoring/src/export.php [hunk 4]
```diff
@@ -243,51 +243,48 @@
...
-            $idC .= ",";
-        }
-        $idC .= $d1['sc_id'];
+        $labels .= '\'' . trim($p) . '\'';
     }
-    $query .= " AND cv2.`value` IN ($idC) ";
+    $servicecategoriesIdCondition = <<<SQL
+s.service_id IN (
+    SELECT service_service_id 
+    FROM {$conf_centreon['db']}.service_categories_relation
```
Comment (sc979) [inline service-monitoring/src/export.php]: is '{$conf_centreon['db']}' mandatory ? and if so why didn't you add it on line 262 ?

File: service-monitoring/src/index.php [hunk 5]
```diff
@@ -294,44 +294,26 @@
...
-        AND h.host_id = s.host_id 
-        WHERE sg.servicegroup_id = :servicegroup_id"
-    );
-    $resultHost->bindValue(':servicegroup_id', $preferences['servicegroup'], PDO::PARAM_INT);
-    $resultHost->execute();
-
-    $hosts = [];
-    while ($row = $resultHost->fetch()) {
-        $hosts[] = $row['host_id'];
+    $resultsSG = explode(',', $preferences['servicegroup']);
```
Comment (sc979) [inline service-monitoring/src/index.php]: same comments for this file ;)

File: service-monitoring/src/index.php [hunk 6]
```diff
@@ -294,68 +294,59 @@
...
+    $querySG ='';
+    foreach ($resultsSG as $resultSG) {
+        if ($querySG != '') {
+            $querySG .= ', ';
```
Comment (jdelpierre) [inline service-monitoring/src/index.php]: In the last instance there's going to be that, is that intended?

**Pull request closed without merging**